packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -1186,8 +1220,13 @@ export class AngularCompilerPlugin {
       ...this.getResourceDependencies(this._compilerHost.denormalizePath(resolvedFileName)),
     ].map((p) => p && this._compilerHost.denormalizePath(p)));
 
-    return [...uniqueDependencies]
-      .filter(x => !!x) as string[];
+    // Add used source files
+    this._usedSourceFiles.add(forwardSlashPath(fileName));
```

Comment:
Somewhat reluctant on adding to the list here. I agree that it's a convenient place to hook into, but it adds somewhat unrelated things to the mental model for this call. 

Would this approach work as well if added to `getCompiledFile` instead (assuming adding imports is not needed either)?

Reply:
Agreed

Reply:
Just checked and `getCompiledFile` will not be called for all files, such as type only files.

Reply:
What about adding an additional Webpack compilation hook inside the Angular plugin that compares the set of TS source files with webpack modules?  Not sure of the appropriate hook off the top of my head though.

Reply:
That’s sound like a good idea I think the hook was called `finishedModules` or something like that. I’ll look into it tmr.

Reply:
Though thinking a bit I am not sure that this will work since I don't think type only files will be part of the webpack modules. Will double check tmr.

Reply:
@clydin Thinking a bit I am not sure that this will work since I don't think type only files will be part of the Webpack modules. Will double check tmr.

Reply:
@clydin I tried the above, but the problem is that type only files will not be part of the webpack modules.

Reply:
Managed to get the information needed through buildInfo.